frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -480,158 +482,291 @@ function ReaderBookmark:onShowBookmark(match_table)
+            local more_selections_enabled = self.select_count < #item_table
+            if actions_enabled then
+                dialog_title = T(N_("1 bookmark selected", "%1 bookmarks selected", self.select_count), self.select_count)
+            else
+                dialog_title = _("No bookmarks selected")
+            end
+            table.insert(buttons, {
+                {
+                    text = _("Select all"),
+                    enabled = more_selections_enabled,
+                    callback = function()
+                        UIManager:close(bm_dialog)
+                        for _, v in ipairs(item_table) do
+                            v.dim = true
+                        end
+                        self.select_count = #item_table
+                        bm_menu:updateItems()
+                    end,
+                },
+                {
+                    text = _("Select page"),
+                    enabled = more_selections_enabled,
+                    callback = function()
+                        UIManager:close(bm_dialog)
+                        local item_first = (bm_menu.page - 1) * bm_menu.perpage + 1
+                        local item_last = math.min(item_first + bm_menu.perpage - 1, #item_table)
+                        for i = item_first, item_last do
+                            if item_table[i].dim == nil then
+                                item_table[i].dim = true
+                                self.select_count = self.select_count + 1
+                            end
+                        end
+                        bm_menu:updateItems()
+                    end,
+                },
+            })
+            table.insert(buttons, {
+                {
+                    text = _("Deselect all"),
+                    enabled = actions_enabled,
+                    callback = function()
+                        UIManager:close(bm_dialog)
+                        for _, v in ipairs(item_table) do
+                            if v.dim then
+                                v.dim = nil
+                            end
+                        end
+                        self.select_count = 0
+                        bm_menu:updateItems()
+                    end,
+                },
+                {
+                    text = _("Reset"),
+                    enabled = actions_enabled and not bookmark.ui.highlight.select_mode,
+                    callback = function()
+                        UIManager:show(ConfirmBox:new{
+                            text = _("Reset page number / timestamp?"),
+                            ok_text = _("Reset"),
+                            ok_callback = function()
+                                UIManager:close(bm_dialog)
+                                for _, v in ipairs(item_table) do
+                                    if v.dim then
+                                        bookmark:removeBookmark(v, true)
```

Comment:
Please add a comment at end of this line: `-- reset_auto_text_only=true`, just so it's clear here that we're doing only that, and no need to go see that function for what `true` means.

It's a bit unclear from the dialog what Reset will do - but ok, the next dialog says more: `Reset page number / timestamp?`
It this something a user would do on individually selected bookmarks? Or something that he would usually do only once to clean "old" bookmarks auto text?
Does it serve any other purpose? If not, could it be made disabled when it doesn't apply and enabled only if we can notice its our old crap with hard-added page number and timestamp ?

Reply:
Reset can be applied individually to any bookmark(s) if a user wants to keep added page number / timestamp in other bookmarks. To reset all bookmarks, just select all and reset.
Added disabling Reset when `bookmarks_items_auto_text` is enabled.

Reply:
> Added disabling Reset when bookmarks_items_auto_text is enabled.

When bookmarks_items_auto_text is disabled, can't you also detect if it would have any effect? And if no, disable Reset?

Reply:
> can't you also detect if it would have any effect?

It requires to scan all bookmarks' `text` field.

Reply:
And to find selected bookmarks we need to scan all bookmarks.

Reply:
Ah, ok, we only show Reset when in select mode, so, well, that's rare :)
Btw, why don't we need to have it in single bookmark long-press? Just because it's rare and best to only make it available in select mode ? (Which I find ok :)
Or because there's another way to reset when in single bookmark mode?
(Sorry, not as deep as you on bookmarks, so may be asking dumb questions :)

Reply:
> Or because there's another way to reset when in single bookmark mode?

Yes, we can save empty note
https://github.com/koreader/koreader/blob/cea870d4f111a398776292c356088a07b5e6db54/frontend/apps/reader/modules/readerbookmark.lua#L993